main.go

```diff
@@ -0,0 +1,323 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"github.com/himanshuo/gameoflife/models"
+	_ "github.com/mattn/go-sqlite3"
+	"html/template"
+	"log"
+	"net/http"
+	"strconv"
+)
+
+//_ is in order to import a package solely for its side-effects at initialization.
+//In this case, go-sqlite3's side effects are allowing sqlite3 to be usable as a
+//database for  sql.Open
+
+//the database resource
+var db *sql.DB
+var Router *mux.Router
+
+//run at start of program.
+func init() {
+
+	startDB()
```

Comment:
startdb here should check whether there was an error or not. If so, do NOT run rest of code.